app/dashboard/models.py

```diff
@@ -2324,6 +2324,9 @@ class UserAction(SuperModel):
     ip_address = models.GenericIPAddressField(null=True)
     location_data = JSONField(default=dict)
     metadata = JSONField(default=dict)
+    utm_campaign = models.CharField(max_length=100, null=True)
```

Comment:
Can you adjust this to simply store `utm` data in a `JSONField`?

Reply:
@mbeacom I have changed utm_campaign to JSON Field, please have a look.